README.md

```diff
@@ -1,3 +1,31 @@
 # Paperkite Github Actions shared actions & workflows
 
 A repository for sharing GHA actions & workflows between projects at PK.
+
+## Actions
+
+### `generate-sha`
+
+Allows us to consistently create 1-to-1 image names for our Docker images via providing a consistent SHA. It's simply an action that provides the 8 char SHA of the current commit to be used in qa/dev tags like dev-81db02f5.
+
+### `generate-environment-info`
+
+This looks at the workflow and determines the environment specific information like:
+
+- the name of the environment: `development|QA|UAT|staging|production`
```

Comment:
Ahh, staging is actually typically an environment that you test your deployments on before going to production. However, in our usage, it's basically a combined dev-QA environment. So some projects have staging instead of dev and QA.

The etymology is from space rocketry:

> Staging is the combination of several rocket sections, or stages, that fire in a specific order and then detach, so a ship can penetrate Earth’s atmosphere and reach space.
>
> The operative principle behind rocket stages is that you need a certain amount of thrust to get above the atmosphere, and then further thrust to accelerate to a speed fast enough to stay in orbit around Earth (orbital speed, about five miles per second).
>
> It’s easier for a rocket to get to that orbital speed without having to carry the excess weight of empty propellant tanks and early-stage rockets. So when the fuel/oxygen for each stage of a rocket is used up, the ship jettisons that stage, and it falls back to Earth. This becomes part of the rocket’s mass fraction—the portion of its fully fueled pre-launch mass that does reach orbit.